src/nih_wayfinding/app/scripts/profiles/profile-model.js

```diff
@@ -8,20 +8,131 @@
     'use strict';
 
     /* ngInject */
-    function ProfileProvider () {
-        var Profile = function() {};
-        Profile.prototype.username = '';
+    function ProfileProvider (localStorageService) {
+        var Profile = function(obj) {
+            if (obj.username) {
+                this.username = obj.username;
+                this.locations = obj.locations || [];
+                this.preferences = obj.preferences || {};
+                this.tempLocation = obj.tempLocation || null;
+                this.transient = false;
+            } else {
+                this.username = '';
+                this.locations = [];
+                this.preferences = {};
+                this.tempLocation = obj.tempLocation || null;
+                this.transient = true;
+            }
+            return this;
+        };
+
+        /**
+         * Governs the persistence in localstorage of the object on which this method is called
+         */
+        Profile.prototype.save = function() {
+            if (this.transient && localStorageService.get(this.username)) {
+                return false;
+            }
+            this.transient = false;
+            localStorageService.set(this.username, this);
+            return true;
+        };
+
+        /**
+         * Sets a user preference
+         *
+         * @param property {string} key
+         * @param value {any} value
+         */
+        Profile.prototype.setPreference = function(property, value) {
```

Comment:
Hmm, I'm a bit torn on these setters. Typically since the objects you're setting are publicly available, they can just be set directly. But, the extra methods are good for discoverability and consistency, and are a bit more future proof, in case other things had to happen here.